docs/source/daml/reference/contract-keys.rst

```diff
@@ -19,76 +19,76 @@ Here's an example of setting up a contract key for a bank account, to act as a b
 
 Because different templates can use the same key type, you need to specify the type of the contract you are trying to fetch using the ``@ContractType`` syntax.
 
 .. _lookupbykey:
 
-``lookupByKey``
+lookupByKey
 ===============
 
 ``optionalContractId <- lookupByKey @ContractType contractKey``
 
 Use ``lookupByKey`` to check whether a contract with the specified key exists. If it does exist, ``lookupByKey`` returns the ``Some contractId``, where ``contractId`` is the ID of the contract; otherwise, it returns ``None``.
 
-``lookupByKey`` needs to be authorized by **all** maintainers of the
-contract you are trying to lookup. There are no restrictions on the
-submitter.
+``lookupByKey`` needs to be authorized by **all** maintainers of the contract you are trying to lookup. The reason for that is denial of service protection. WIthout this restriction, a malicious Party could legitimately request another party to validate any number of negative contract key lookups without that other party ever having signed anything.
 
-If the lookup fails (ie returns ``None``), this guarantees that no contract has this key.
+Unlike ``fetchByKey``, the transaction **does not fail** if no contract with the given key is found. Instead, ``lookupByKey`` just returns ``None``. However, that does not guarantee that no such contract exists. The submitting party may simply not know about it.
+
+``lookupByKey`` fails and aborts the transaction if:
+
+- Authorization from at least one maintainer is missing. This check fails at interpretation time.
+- The lookup is incorrect. This can happen either due to contention, or because the submitter didn't know of the contract. This check fails at validation time.
 
-Unlike ``fetchByKey``, the transaction **does not fail** if a contract with the key doesn't exist: instead, ``lookupByKey`` just returns ``None``.
 
 To get the data from the contract once you've confirmed it exists, you'll still need to use ``fetch``.
 
 Because different templates can use the same key type, you need to specify the type of the contract you are trying to fetch using the ``@ContractType`` syntax.
 
-``exerciseByKey``
+exerciseByKey
 =================
 
 ``exerciseByKey @ContractType contractKey``
 
-Use ``exerciseByKey`` to exercise a choice on a contract identified by its ``key`` (compared to ``exercise``, which lets you exercise a contract identified by its ``ContractId``). To run ``exerciseByKey`` you need authorization from the controllers of the choice and at least one of the key maintainers.
+Use ``exerciseByKey`` to exercise a choice on a contract identified by its ``key`` (compared to ``exercise``, which lets you exercise a contract identified by its ``ContractId``). To run ``exerciseByKey`` you need authorization from the controllers of the choice and at least one stakeholder.
```

Comment:
Maybe worth pointing out that the restrictions are the same as the one for `fetchByKey` + `exerciseByKey`?

Reply:
@cocreature probably means `fetchByKey + exercise`. Referring to that combination is a great idea.

Reply:
:facepalm: yeah I definitely mean `exercise` not `exerciseByKey`